packages/gnarly-core/src/models/Transaction.ts

```diff
@@ -37,7 +46,33 @@ export default class Transaction {
     this.methodName = methodAbi.fullName
     this.signature = methodAbi.signature
     this.methodId = methodAbi.shortId
-    // @TODO(shrugs) - do that
-    this.args = {}
+
+    // get abi item
+    const paramAbiItem = registeredAbi.find((item) => item.signature === this.signature)
+    if (!paramAbiItem) { return }
+    // ^ using incorrect abi
+
+    const parameterTypes = paramAbiItem.inputs.filter((item) => ({
```

Comment:
what's the purpose of the filter here? should this be a `map`?

Reply:
yep, probably not needed at all! Will push update after I test it, I was going for a `map` tho 'o.o